src/components/markdownEditor/index.tsx

```diff
@@ -0,0 +1,208 @@
+import Indent from "../../../public/icons/indent.svg";
+import Underline from "../../../public/icons/underline.svg";
+import Videocam from "../../../public/icons/videocam.svg";
+import LeftAlign from "../../../public/icons/left-align.svg";
+import RightAlign from "../../../public/icons/right-align.svg";
+import CenterAlign from "../../../public/icons/center-align.svg";
+
+const underlineCommand: ICommand = {
+    name: "underline",
+    keyCommand: "underline",
+    shortcuts: "ctrlcmd+u",
+    buttonProps: { "aria-label": "Add underline text" },
+    icon: <Underline />,
+    execute: (state: TextState, api: TextAreaTextApi) => {
+        // Adjust the selection to encompass the whole word if the caret is inside one
+        const newSelectionRange = selectWord({
+            text: state.text,
+            selection: state.selection,
+        });
+        const state1 = api.setSelectionRange(newSelectionRange);
+        // Replaces the current selection with the italic mark up
+        const state2 = api.replaceSelection(`__${state1.selectedText}__`); // convention to notate underline: double underscore
+        // Adjust the selection to not contain the __
+        api.setSelectionRange({
+            start: state2.selection.end - 2 - state1.selectedText.length,
+            end: state2.selection.end - 2,
+        });
+    },
+};
+
+const videoCommand: ICommand = {
+    name: "video",
+    keyCommand: "video",
+    shortcuts: "ctrlcmd+y",
+    buttonProps: { "aria-label": "Add video", title: "Add video" },
+    icon: <Videocam />,
+    execute: (state: TextState, api: TextAreaTextApi) => {
+        // Select everything
+        const newSelectionRange = selectWord({
+            text: state.text,
+            selection: state.selection,
+        });
+        const state1 = api.setSelectionRange(newSelectionRange);
+        // Replaces the current selection with the video
+        const videoTemplate =
+            state1.selectedText || "https://youtube.com/your-video-link";
+        api.replaceSelection(`$[](${videoTemplate})`); // convention: video links are prepended with "$" instead of "!"
+        // Adjust the selection to not contain the **
+        api.setSelectionRange({
+            start: 4 + state1.selection.start,
+            end: 4 + state1.selection.start + videoTemplate.length,
+        });
+    },
+};
+
+const indentCommand: ICommand = {
+    name: "indent",
+    keyCommand: "indent",
+    shortcuts: "ctrlcmd+m",
+    buttonProps: { "aria-label": "Indent text", title: "Indent" },
+    icon: <Indent />,
+    execute: (state: TextState, api: TextAreaTextApi) => {
+        let modifyText = `$ ${state.selectedText}\n`; // convention: notate indent with $
+        if (!state.selectedText) {
+            modifyText = `$ `;
+        }
+        api.replaceSelection(modifyText);
+    },
+};
+
+const rightAlignCommand: ICommand = {
+    name: "right-align",
+    keyCommand: "right-align",
+    shortcuts: "ctrlcmd+r",
```

Comment:
not sure how this short cut would work since `Cmd+r` is already refresh, right?

Reply:
it'll override the default behavior (e.g. refresh) if you're within that editor, though yeah that could be confusing